server/webapp/WEB-INF/rails.new/webpack/views/dashboard/models/dashboard_view_model.js

```diff
@@ -47,24 +47,41 @@ const VM = () => {
     },
 
     dropdown: {
-      isDropDownOpen: (name) => pipelinesState[name][DROPDOWN_KEY](),
+      isDropDownOpen: (name, instanceLabel) => pipelinesState[name][DROPDOWN_KEY][instanceLabel](),
 
-      toggle: (name) => {
-        viewModel.dropdown.hideAllExcept(name);
-        pipelinesState[name][DROPDOWN_KEY](!pipelinesState[name][DROPDOWN_KEY]());
+      toggle: (name, instanceLabel) => {
+        viewModel.dropdown.hideAllExcept(name, instanceLabel);
+        pipelinesState[name][DROPDOWN_KEY][instanceLabel](!pipelinesState[name][DROPDOWN_KEY][instanceLabel]());
       },
 
-      hideAllExcept: (name) => {
-        _.each(_.keys(pipelinesState), (key) => {
-          if (key !== name) {
-            pipelinesState[key][DROPDOWN_KEY](false);
+      hideDropDownForOtherInstances: (name, instanceLabel) => {
+        _.each(_.keys(pipelinesState[name][DROPDOWN_KEY]), (label) => {
```

Comment:
can we directly use `_.each(pipelinesState[name][DROPDOWN_KEY], (val, key) => {})`?
here we are iterating over the collection twice.